app/controllers/search_controller.rb

```diff
@@ -54,16 +54,15 @@ def load_vars
     if @splited_query.first.length < MIN_QUERY_CHARS
       flash[:error] = t'search.index.error.query_length', n: MIN_QUERY_CHARS
       redirect_to :back
-    else
+    elsif @splited_query.length > 1
+      @search_query = ''
       @splited_query.each_with_index do |w, i|
-        if @splited_query.length > 1 && w.length < MIN_QUERY_CHARS
-          @splited_query.slice!(i)
-        end
+        @search_query += "#{@splited_query[i]} " if w.length >= MIN_QUERY_CHARS
```

Comment:
I'm not sure, is this safe enough? Looks like an easy way to perform SQL injection.